pkg/client/restclient/client.go

```diff
@@ -46,7 +46,7 @@ const (
 //
 // Most consumers should use client.New() to get a Kubernetes API client.
 type RESTClient struct {
-	urlProvider func() *url.URL
+	urlProvider *URLProvider
```

Comment:
> EDIT: Should just be an interface, not a pointer to an interface (urlProvider URLProvider)

ping @dshulyak (since editing comments doesn't create notifications)